### ISSUE TITLE
Extra space in generic operators

### ISSUE THREAD

**Issue by bencochran** [nicklockwood/SwiftFormat#52]
Given the following before formatting:
```swift
func ==<T>(left: T, right: T) -> Bool {
    return true
}
```

I get this after formatting:
```swift
func ==<T > (left: T, right: T) -> Bool {
    return true
}
```

**Comment by bencochran** [comment]
I started exploring a fix: https://github.com/nicklockwood/SwiftFormat/compare/master...bencochran:generic-operator

It handles the above case, but would still fail with:
```swift
func ==<S: Sequence where S.Iterator.Element == ()>(left: T, right:T) -> Bool {
    return false
}
```
(which will be invalid syntax in future Swift, but for now…)

**Comment by nicklockwood** [comment]
Thanks for trying! I ended up doing quite a major refactor to fix this.

My original implementation was treating < and > as operators and then trying to work out if they were generic scope delimiters, which meant I was having to both split up operators like >> and <<=, *and* stitch them back together again if it turned out that a > that I thought was a generic was really a greater-than operator.

In the new implementation, I assume that they are scope delimiters initially, and then convert them to operators if they turn out not to be. This is much simpler as I no longer have to worry about splitting operators, only merging them.

The fix is included in 0.16.2.

**Comment by bencochran** [comment]
Amazing, thank you! I wasn’t a big fan of the approach I was taking for this fix, but it was a good tiptoe into how your tokenizer works. The real fix looks nice!
